api/src/pcapi/core/offers/api.py

```diff
@@ -1296,3 +1298,33 @@ def delete_price_category(offer: models.Offer, price_category: models.PriceCateg
     validation.check_price_categories_deletable(offer)
     db.session.delete(price_category)
     db.session.commit()
+
+
+def list_nearest_venues_for_offer(
+    offer: models.Offer,
+    latitude: float | None = None,
+    longitude: float | None = None,
+) -> list[offerers_models.Venue]:
+    if latitude is not None and longitude is not None:
+        center = func.ST_MakePoint(longitude, latitude)
+    else:
+        center = func.ST_MakePoint(offer.venue.longitude, offer.venue.latitude)
+    venues = (
+        offerers_models.Venue.query.join(offerers_models.Venue.offers)
+        .join(offerers_models.Venue.managingOfferer)
+        .options(contains_eager(offerers_models.Venue.managingOfferer))
+        .filter(models.Offer.id != offer.id)
+        .filter(offerers_models.Offerer.isActive)
+        .filter(offerers_models.Offerer.isValidated)
+        .filter(models.Offer.released)
+        .filter(~models.Offer.isSoldOut)  # type: ignore [operator] # pylint: disable=invalid-unary-operand-type
+        .filter(models.Offer.extraData["isbn"].astext == offer.extraData["isbn"])  # type: ignore [index]
+        .order_by(
+            func.ST_Distance(
+                func.Geometry(func.ST_MakePoint(offerers_models.Venue.longitude, offerers_models.Venue.latitude)),
+                func.Geometry(center),
+            )
+        )
+        .limit(10)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Pour la suite, ce serait top de pouvoir paginer au lieu de simplement limiter à 10